stdlib/public/FreeBSDlibc/FreeBSDlibc.swift

```diff
@@ -0,0 +1,223 @@
+
+public var S_IRWXU: mode_t  { return mode_t(0o000700) }
+public var S_IRUSR: mode_t  { return mode_t(0o000400) }
+public var S_IWUSR: mode_t  { return mode_t(0o000200) }
+public var S_IXUSR: mode_t  { return mode_t(0o000100) }
+
+public var S_IRWXG: mode_t  { return mode_t(0o000070) }
+public var S_IRGRP: mode_t  { return mode_t(0o000040) }
+public var S_IWGRP: mode_t  { return mode_t(0o000020) }
+public var S_IXGRP: mode_t  { return mode_t(0o000010) }
+
+public var S_IRWXO: mode_t  { return mode_t(0o000007) }
+public var S_IROTH: mode_t  { return mode_t(0o000004) }
+public var S_IWOTH: mode_t  { return mode_t(0o000002) }
+public var S_IXOTH: mode_t  { return mode_t(0o000001) }
+
+public var S_ISUID: mode_t  { return mode_t(0o004000) }
+public var S_ISGID: mode_t  { return mode_t(0o002000) }
+public var S_ISVTX: mode_t  { return mode_t(0o001000) }
+
+//===----------------------------------------------------------------------===//
+// signal.h
+//===----------------------------------------------------------------------===//
+
+#if os(FreeBSD)
+public var SIG_DFL: __sighandler_t? { return nil }
+public var SIG_IGN: __sighandler_t {
+  return unsafeBitCast(1, __sighandler_t.self)
+}
+public var SIG_ERR: __sighandler_t {
+  return unsafeBitCast(-1, __sighandler_t.self)
+}
+public var SIG_HOLD: __sighandler_t {
+  return unsafeBitCast(2, __sighandler_t.self)
+}
+#endif
+
+//===----------------------------------------------------------------------===//
+// semaphore.h
+//===----------------------------------------------------------------------===//
+
+/// The value returned by `sem_open()` in the case of failure.
+public var SEM_FAILED: UnsafeMutablePointer<sem_t> {
+  // The value is ABI.  Value verified to be correct on FreeBSDlibc.
+  return UnsafeMutablePointer<sem_t>(bitPattern: 0)
+}
+
+@warn_unused_result
+@_silgen_name("_swift_FreeBSDlibc_sem_open2")
+internal func _swift_FreeBSDlibc_sem_open2(
+  name: UnsafePointer<CChar>,
+  _ oflag: CInt
+) -> UnsafeMutablePointer<sem_t>
+
+@warn_unused_result
+@_silgen_name("_swift_FreeBSDlibc_sem_open4")
+internal func _swift_FreeBSDlibc_sem_open4(
+  name: UnsafePointer<CChar>,
+  _ oflag: CInt,
+  _ mode: mode_t,
+  _ value: CUnsignedInt
+) -> UnsafeMutablePointer<sem_t>
+
+@warn_unused_result
+public func sem_open(
+  name: UnsafePointer<CChar>,
+  _ oflag: CInt
+) -> UnsafeMutablePointer<sem_t> {
+  return _swift_FreeBSDlibc_sem_open2(name, oflag)
+}
+
+@warn_unused_result
+public func sem_open(
+  name: UnsafePointer<CChar>,
+  _ oflag: CInt,
+  _ mode: mode_t,
+  _ value: CUnsignedInt
+) -> UnsafeMutablePointer<sem_t> {
+  return _swift_FreeBSDlibc_sem_open4(name, oflag, mode, value)
+}
+
+// extern char **environ;
+@_silgen_name("_swift_FreeBSDlibc_getEnvironment")
+internal func _swift_FreeBSDlibc_getEnvironment(
+) -> UnsafeMutablePointer<UnsafeMutablePointer<CChar>>
+
+@warn_unused_result
+public func freebsdGetEnviron() -> UnsafeMutablePointer<UnsafeMutablePointer<CChar>> {
```

Comment:
If `extern char **environ` is the API on FreeBSD that we don't import, then the overlay should be exposing a readonly var:

```
var environ: UnsafeMutablePointer<UnsafeMutablePointer<CChar>> {
  ...
}
```